src/mono/wasi/runtime/driver.c

```diff
@@ -761,36 +661,63 @@ mono_wasm_string_get_data (
 }
 
 void add_assembly(const char* base_dir, const char *name) {
-    FILE *fileptr;
-    unsigned char *buffer;
-    long filelen;
-    char filename[256];
-    sprintf(filename, "%s/%s", base_dir, name);
-    // printf("Loading %s...\n", filename);
-
-    fileptr = fopen(filename, "rb");
-    if (fileptr == 0) {
-        printf("Failed to load %s\n", filename);
-        fflush(stdout);
-    }
-
-    fseek(fileptr, 0, SEEK_END);
-    filelen = ftell(fileptr);
-    rewind(fileptr);
-
-    buffer = (unsigned char *)malloc(filelen * sizeof(char));
-    fread(buffer, filelen, 1, fileptr);
-    fclose(fileptr);
-
-    assert(mono_wasm_add_assembly(name, buffer, filelen));
+	FILE *fileptr;
+	unsigned char *buffer;
+	long filelen;
+	char filename[256];
+	sprintf(filename, "%s/%s", base_dir, name);
+	// printf("Loading %s...\n", filename);
+
+	fileptr = fopen(filename, "rb");
+	if (fileptr == 0) {
+		printf("Failed to load %s\n", filename);
+		fflush(stdout);
+	}
+
+	fseek(fileptr, 0, SEEK_END);
+	filelen = ftell(fileptr);
+	rewind(fileptr);
+
+	buffer = (unsigned char *)malloc(filelen * sizeof(char));
+	if(!fread(buffer, filelen, 1, fileptr)) {
+		printf("Failed to load %s\n", filename);
+		fflush(stdout);
+	}
+	fclose(fileptr);
+
+	assert(mono_wasm_add_assembly(name, buffer, filelen));
 }
 
 MonoMethod* lookup_dotnet_method(const char* assembly_name, const char* namespace, const char* type_name, const char* method_name, int num_params) {
-    MonoAssembly* assembly = mono_wasm_assembly_load (assembly_name);
+	MonoAssembly* assembly = mono_wasm_assembly_load (assembly_name);
 	assert (assembly);
-    MonoClass* class = mono_wasm_assembly_find_class (assembly, namespace, type_name);
+	MonoClass* class = mono_wasm_assembly_find_class (assembly, namespace, type_name);
 	assert (class);
-    MonoMethod* method = mono_wasm_assembly_find_method (class, method_name, num_params);
+	MonoMethod* method = mono_wasm_assembly_find_method (class, method_name, num_params);
 	assert (method);
 	return method;
 }
+
+int main() {
+	// Assume the runtime pack has been copied into the output directory as 'runtime'
+	// Otherwise we have to mount an unrelated part of the filesystem within the WASM environment
+	mono_set_assemblies_path(".:./runtime/native:./runtime/lib/net7.0");
```

Comment:
why net7.0?

Reply:
because main branch is still net7. I will replace it with symbol.

Reply:
I removed this on my branch. This isn't really needed. I mean, it can be any path like we have `managed/` in wasm. But also, this isn't needed when we bundle everything into a single file.

Reply:
Long term I prefer pre-loading to memory instead of doing it as "VFS". But for this PR, I prefer not bundling it at all.
I made it `net8.0` now as I was not able to quickly figure out how to pass it via cmake.
@radical do you know what part of the build influnces this `/runtime/lib/net8.0` folder nesting ?

Reply:
AFAICS, this is a completely arbitrary path. And removed it on my branch for relinking case. We can choose whatever path, and populate that from the build.